src/ScheduleService.Data/WorkspaceRepository.cs

```diff
@@ -1,41 +1,131 @@
 ﻿using LT.DigitalOffice.ScheduleService.Data.Interfaces;
+using LT.DigitalOffice.ScheduleService.Data.Provider;
 using LT.DigitalOffice.ScheduleService.Models.Db;
+using LT.DigitalOffice.ScheduleService.Models.Dto.Requests.Workspace;
+using LT.DigitalOffice.ScheduleService.Models.Dto.Requests.Workspace.Filters;
 using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.EntityFrameworkCore;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.ScheduleService.Data;
 
 public class WorkspaceRepository : IWorkspaceRepository
 {
-  public Task CreateAsync(DbWorkspace board)
+  private readonly IDataProvider _provider;
+
+  public WorkspaceRepository(
+    IDataProvider provider)
   {
-    throw new NotImplementedException();
+    _provider = provider;
   }
 
-  public Task<bool> EditByIdAsync(Guid id, JsonPatchDocument<DbWorkspace> dbWorkspace)
+  public async Task<Guid?> CreateAsync(DbWorkspace workspace)
```

Comment:
just guid. without ?

Reply:
Guid? is in order to body will be null in case of invalid request